submissions/Ant-C-tech/oop-exercise/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(name, gender, friends) {
+        //At the moment I can't imagine what the creatures will inhabit our world, that is why I assumed that these will be some people and different kinds of their pets. In this way I can assume that they all would have these common properties and one method to perform the task.
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+        this.templete = ` My name is ${this.name}. My gender is ${this.gender}.`;
+    };
+
+    getVoice() {
+        let pluralEnding;
+        (this.friends && this.friends.length > 1) ? pluralEnding = 's': pluralEnding = '';
+        return (this.friends) ? this.templete + `I have friend${pluralEnding}: ${this.friends.join(', ')}.` : this.templete + ' I am looking for friends.';
+    };
+};
+
+class Animal extends Inhabitant {
+    constructor(...args) {
+        super(...args);
+        this.legs = 4;
+    };
+
+    getVoice() {
+        return super.getVoice() + ` I have ${this.legs} legs.`;
+    };
+}
+
+class Human extends Inhabitant {
+    constructor(saying, ...args) {
+        super(...args);
+        //I assumed that people can say different phrases
+        this.saying = saying;
+        this.species = 'human';
+        this.hands = 2;
+        this.legs = 2;
+    };
+
+    getVoice() {
+        return `I am a ${this.species}. ` + super.getVoice() + ` I have ${this.legs} legs and ${this.hands} hands. I want to say you: "${this.saying}".`;
+    };
```

Comment:
1. If you had more optimal classes design you might benefit from method overloading.
```
methodName() {
  return [super.methodName(), 'bar'].join(", "); // calls methodName from a parent class
}
```

2. Lengthy strings are optimizable. Example:
```
[ `My name is ${name}`,
  'I am X yrs old',
  'My friends are',
].join(". ");
```
This also makes maintenance easier. If we want to add a new fragment, we can easily spot a right place for it.